apps/site/lib/site_web/controllers/trip_plan_controller.ex

```diff
@@ -96,6 +102,49 @@ defmodule SiteWeb.TripPlanController do
     %{leg | mode: mode_with_fares}
   end
 
+  @spec highest_highest_month_pass(Itinerary.t()) :: Fare.t() | nil
+  defp highest_highest_month_pass(%Itinerary{legs: legs}) do
+    legs
+    |> Enum.map(&highest_month_pass/1)
+    |> max_by_cents()
+  end
+
+  @spec highest_lowest_month_pass(Itinerary.t()) :: Fare.t() | nil
+  defp highest_lowest_month_pass(%Itinerary{legs: legs}) do
```

Comment:
I don't really get the naming of these two functions... help? 😅 
I'm also starting to wonder whether we should migrate away from using "highest" and "lowest" to something more meaningful.

Reply:
Yeah, I've been struggling with all these names the whole time. It's tricky because you can't necessarily go by media, they can mean different things for different modes and contexts. Maybe we can use the "base" and "recommended" terms we present to the users. I was trying to avoid those because we used to use "base" in the codebase to mean what we are now calling "recommended" which is obviously confusing. But we removed that and if we decide that it's not going to confuse any of us, maybe that route is best? So these functions could be something like `highest_base_monthly_pass` and `highest_recommended_monthly_pass`. Or do you have any other recommendations?

I'm looking forward to the day when these fare differences just go away!

Reply:
@thecristen Or maybe another approach would be to just change the first "highest". I don't know what the best alternative would be, but maybe instead these could be something like `overarching_highest_month_pass` and `overarching_lowest_month_pass`?

Reply:
Yeah, I see the struggle! I think introducing some consistency between the backend and frontend would make the _most_ sense. I'd go more conceptual rather than using the function name to explain the implementation. IMO `highest_base_monthly_pass` and `highest_recommended_monthly_pass` wouldn't make sense here are there will only be one recommended monthly pass, and I'm not sure "base" is a relevant concept for monthly passes anyway (which have no reduced fare counterpart AFAIK).

We hadn't discussed it much, but my overarching understanding is:

**For a trip:**
Base = sum of all the regular one-way fares involved
Reduced  = same, but with reduced fare instead of full fare values
(eventually these will incorporate transfers, which will lower the resulting values)

**For monthly passes:**
All monthly passes relevant to the trip (there may be more than one) will be shown in the accordion - includes regular (base?) fares as well as reduced fares
One of these passes will be highlighted as the Recommended one (the highest one, maybe?).

Sorry I don't have a simpler answer at the moment but hopefully this aids our thinking. This discussion all applies to the functions in the `Month` and `OneWay` modules too.

Reply:
I think it's actually a little more complicated than that for passes. 

Each leg of your trip will have potential passes for it. Let's take an example where you take the subway to the commuter rail. There is one monthly pass for the subway (Monthly LinkPass). It's good on the subway and bus but it doesn't get you on the CR. For the CR there are two monthly passes available, the CharlieTicket or the mTicket. The CR pass does grant you access to the subway. 

So this is the reason for the two `highest`s in `highest_highest_month_pass`. The first (inner) one is because we need to figure out the highest or base fare for each mode. The Monthly LinkPass in the case of the subway, and the CharlieTicket in the case of CR. And then the second `highest` is because we need to say, ok, given all these legs, what pass grants you access to everything? So, in this case, we want to display the CR CharlieTicket monthly pass, not the subway Monthly LinkPass.

Reply:
Thanks for the explanation! I think that's not too different than what I was thinking. My thoughts were more: first get _every_ pass for every mode (e.g. [LinkPass, CR CharlieTicket,  CR mTicket]) and then pick the recommended one via a view function (e.g. CR CharlieTicket). It'd be clearer and we can use the list of all passes in the accordion later.

Reply:
Ah that’s an interesting thought. So with that approach we’d need to figure out which passes to display or not display later? With my current approach I was thinking we’d always display the highest and display the lowest only if it was different. I’m trying to think about what the logic would be with a list. Maybe going through the covered modes and see what covers everything?

Reply:
Maybe! If that's too complicated, we might want to put that off until a future PR. 

But all this was to say that your original function names still confused me and I wouldn't have been sure how to leverage them if I were to pick up related tasks :/